src/QMCDrivers/tests/test_QMCDriverFactory.cpp

```diff
@@ -57,34 +62,70 @@ TEST_CASE("QMCDriverFactory create VMC Driver", "[qmcapp]")
   qmc_driver =
       driver_factory.createQMCDriver(node, das, *qmc_system, particle_pool, wavefunction_pool, hamiltonian_pool, comm);
   REQUIRE(qmc_driver != nullptr);
+  REQUIRE_THROWS(dynamic_cast<VMCBatched&>(*qmc_driver));
+  REQUIRE_NOTHROW(dynamic_cast<VMC&>(*qmc_driver));
+  CHECK(qmc_driver->getEngineName() == "VMC");
 }
 
 TEST_CASE("QMCDriverFactory create VMCBatched driver", "[qmcapp]")
 {
-  using namespace testing;
   Communicate* comm;
   comm = OHMMS::Controller;
+  using namespace testing;
 
-  ProjectData test_project;
-  QMCDriverFactory driver_factory(test_project);
-
-  Libxml2Document doc;
-  bool okay = doc.parseFromString(valid_vmc_input_sections[valid_vmc_input_vmc_batch_index]);
-  REQUIRE(okay);
-  xmlNodePtr node                           = doc.getRoot();
-  QMCDriverFactory::DriverAssemblyState das = driver_factory.readSection(node);
-  REQUIRE(das.new_run_type == QMCRunType::VMC_BATCH);
-
-  auto particle_pool     = MinimalParticlePool::make_diamondC_1x1x1(comm);
-  auto wavefunction_pool = MinimalWaveFunctionPool::make_diamondC_1x1x1(comm, particle_pool);
-  auto hamiltonian_pool  = MinimalHamiltonianPool::make_hamWithEE(comm, particle_pool, wavefunction_pool);
-  std::string target("e");
-  MCWalkerConfiguration* qmc_system = particle_pool.getWalkerSet(target);
-
-  std::unique_ptr<QMCDriverInterface> qmc_driver;
-  qmc_driver =
-      driver_factory.createQMCDriver(node, das, *qmc_system, particle_pool, wavefunction_pool, hamiltonian_pool, comm);
-  REQUIRE(qmc_driver != nullptr);
+  SECTION("driver epoch behavior")
+  {
+    ProjectData test_project("batched");
+    QMCDriverFactory driver_factory(test_project);
+
+    Libxml2Document doc;
+    bool okay = doc.parseFromString(valid_vmc_input_sections[valid_vmc_input_vmc_batch_index]);
+    REQUIRE(okay);
+    xmlNodePtr node                           = doc.getRoot();
+    QMCDriverFactory::DriverAssemblyState das = driver_factory.readSection(node);
+    REQUIRE(das.new_run_type == QMCRunType::VMC_BATCH);
+
+    auto particle_pool     = MinimalParticlePool::make_diamondC_1x1x1(comm);
+    auto wavefunction_pool = MinimalWaveFunctionPool::make_diamondC_1x1x1(comm, particle_pool);
+    auto hamiltonian_pool  = MinimalHamiltonianPool::make_hamWithEE(comm, particle_pool, wavefunction_pool);
+    std::string target("e");
+    MCWalkerConfiguration* qmc_system = particle_pool.getWalkerSet(target);
+
```

Comment:
Could you move 88-92 outside the section and share them between the two sections? Try to reduce duplicated lines. The same for DMC as well.